### PR TITLE
[backend/frontend] Management panel for restricted entities - chunk1 (#4961)

### PR DESCRIPTION
**Pull request by marieflorescontact**
### Proposed changes

* Add `stixCoreObjectsRestricted` api to list all restricted entities (for BYPASS user only) + tests (+ 2 create + 2 update + 2 delete)
* Add Data > Management > Restricted entites root
* Add DataTable screen to list restricted entities
* Add explanation card
* Add FF for the menus 

### Related issues

* https://github.com/OpenCTI-Platform/opencti/issues/4961
* https://github.com/OpenCTI-Platform/opencti/issues/4538

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality
- [x] I wrote test cases for the relevant uses case (coverage and e2e)
- [ ] I added/update the relevant documentation (either on github or on notion)
- [x] Where necessary I refactored code to improve the overall quality

### CODE REVIEW

File: opencti-platform/opencti-front/src/components/dataGrid/DataTable.tsx
Comment (delemaf) [inline opencti-platform/opencti-front/src/components/dataGrid/DataTable.tsx]: Looks like i18n is missing.
```suggestion
            {t_i18n('This list displays all the entities that have some access restriction enabled, meaning that they are only accessible to some specific users. You can remove this access restriction on this screen.')}
```

File: opencti-platform/opencti-front/src/private/components/data/Management.tsx
Comment (delemaf) [inline opencti-platform/opencti-front/src/private/components/data/Management.tsx]: `isRuntimeFieldEnable` seems to always return a **bool** nullish coalescing operator might be useless.
```suggestion
  const isRuntimeSort = isRuntimeFieldEnable();
```

**Review by delemaf — COMMENTED** [review]
Tested functionally 🚀 Just few little code suggestion

File: opencti-platform/opencti-graphql/src/domain/stixCoreObject.js
Comment (aHenryJard) [inline opencti-platform/opencti-graphql/src/domain/stixCoreObject.js]: Would be better to add detail on restricted, because some entity can be restricted because of marking or organization. Feel free to use another name than my proposal.
```suggestion
export const findAllAuthMemberRestricted = async (context, user, args) => {
```

File: opencti-platform/opencti-graphql/tests/02-integration/02-resolvers/container-authorized-members-test.ts
Comment (aHenryJard) [inline opencti-platform/opencti-graphql/tests/02-integration/02-resolvers/container-authorized-members-test.ts]:
```suggestion
  it('user without bypass should not be allowed list all auth member restricted entities', async () => {
```

File: opencti-platform/opencti-graphql/tests/02-integration/02-resolvers/container-authorized-members-test.ts
Comment (aHenryJard) [inline opencti-platform/opencti-graphql/tests/02-integration/02-resolvers/container-authorized-members-test.ts]:
```suggestion
  it('should BYPASS user be allowed list all auth member restricted entities', async () => {
```

File: opencti-platform/opencti-graphql/tests/02-integration/02-resolvers/container-authorized-members-test.ts
Comment (aHenryJard) [inline opencti-platform/opencti-graphql/tests/02-integration/02-resolvers/container-authorized-members-test.ts]:
```suggestion
  it('should platform organization sharing and EE activated', async () => {
```

File: opencti-platform/opencti-graphql/tests/02-integration/02-resolvers/container-authorized-members-test.ts
Comment (aHenryJard) [inline opencti-platform/opencti-graphql/tests/02-integration/02-resolvers/container-authorized-members-test.ts]:
```suggestion
  it('using platform org - user without bypass should not be allowed list all auth member restricted entities', async () => {
```

File: opencti-platform/opencti-graphql/tests/02-integration/02-resolvers/container-authorized-members-test.ts
Comment (aHenryJard) [inline opencti-platform/opencti-graphql/tests/02-integration/02-resolvers/container-authorized-members-test.ts]:
```suggestion
  it('using platform org - should BYPASS user be allowed list all auth member restricted entities', async () => {
```

File: opencti-platform/opencti-graphql/tests/02-integration/02-resolvers/container-authorized-members-test.ts
Comment (aHenryJard) [inline opencti-platform/opencti-graphql/tests/02-integration/02-resolvers/container-authorized-members-test.ts]:
```suggestion
  it('should platform organization sharing and EE deactivated', async () => {
```

**Review by aHenryJard — CHANGES_REQUESTED** [review]
The data management menu on the left is displayed without feature flag, so I'm blocking for now.

**Comment by CelineSebe** [comment]
All comments are resolved. Waiting for approval.

File: opencti-platform/opencti-front/src/private/components/data/Root.jsx
Comment (SouadHadjiat) [inline opencti-platform/opencti-front/src/private/components/data/Root.jsx]: DATA_MANAGEMENT_RIGHT_MENU

File: opencti-platform/opencti-front/src/private/components/data/Management.tsx
Comment (SouadHadjiat) [inline opencti-platform/opencti-front/src/private/components/data/Management.tsx]: DATA_MANAGEMENT_RIGHT_MENU

**Review by SouadHadjiat — APPROVED** [review]

**Review by aHenryJard — APPROVED** [review]
Removing the change request

**Pull request merged**
